### DISCUSSION TITLE
bluesky_cli limits (max 100) - counting members of a list

### DISCUSSION THREAD

**Discussion by kesma01** [myConsciousness/atproto.dart#981] [Q&A] [answered]
Hello,

i am probably not familiar enough with the AT protocol and was not able to answer the question with what i have read about it.
I have a problem to count the members in a list. Do i need to implement the cursor mechanism and use max limit 100 in the CLI, or the max limit of 100, can be changed somewhere in the configuration of the bluesky_cli?

Any hints or workarounds for counting members of a list are also welcome :)

Thank you and best regards,
Kesma

**Comment by myConsciousness** [comment, marked as answer]
Hi @kesma01!

If you use `bluesky_cli` to count all the members of a list, you should use both limit and cursor.

First, you can get a list of specific ACTOR listings as follows:

```bash
bsky lists --actor=shinyakato.dev
```

You can then use the URI in the `uri` field of the result of the above command to get the members of the list as follows:

```bash
bsky list --uri=at://did:plc:iijrtk7ocored6zuziwmqq3c/app.bsky.graph.list/3katlozdqmh2p
```

To specify limit and cursor in this case, you can do the following:

```bash
bsky list --uri=at://did:plc:iijrtk7ocored6zuziwmqq3c/app.bsky.graph.list/3katlozdqmh2p --limit=100 --cursor=1696324904535::bafyreiafaxy4lhmyyqsx4lxfespwdosm6fy36zqggycrd4z2wzs2ti2zo4
```

---

Let me know if you need a more detailed description for your app's use case!

**Reply by kesma01** [reply]
Thank you for your extensive explanation @myConsciousness! 

I will most probably write a (shell) script to count all members in a specific list, using current limit max of 100.  The cursor mechanism is, as i can see, used all over the place and i will put some effort on some generalisations for other bsky commands. I just need a module that manipulate with cursors.
